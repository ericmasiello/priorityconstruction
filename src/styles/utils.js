/* eslint-disable import/prefer-default-export */
export const pxToRem = (value, base = 16) => `${value / base}rem`;
