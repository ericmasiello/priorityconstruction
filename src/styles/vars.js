import tinyColor from 'tinycolor2';

const unitToPx = unit => unit * 16;
const baseColor = '#424141';
export const scaler = 1.5;

export const COLORS = {
  bg: '#fff',
  base: baseColor,
  muted: '#787575',
  highlight: '#FBDB1E',
  highlight2: '#4D1017',
  highlight3: '#002B6D',
  link: '#18232C',
  border: tinyColor(baseColor).lighten(50).toRgbString(),
};

export const MAX_CONTENT_WIDTH = 1024;

export const PAGE_SPACING = {
  horizontal: 20,
};

// type properties
export const TYPE_SIZE = {
  small: [unitToPx(0.8125), scaler],
  base: [unitToPx(1), scaler],
  t5: [unitToPx(1.3), scaler],
  t4: [unitToPx(1.5), scaler],
  t3: [unitToPx(2), 1.3],
  t2: [unitToPx(2.5), 1.2],
  t1: [unitToPx(3), 1.1],
  jumbo: [unitToPx(4), 1.1],
};

export const BODY_WEIGHTS = {
  regular: 400,
  medium: 500,
  bold: 600,
};

const getNumericWeights = weights => (
  Object.keys(weights)
    .map(key => weights[key])
    .join(',')
);

export const BODY_FONT_STACK = '\'Montserrat\', sans-serif';

export const FONT_URL = [
  'https://fonts.googleapis.com/css?family=',
  `Montserrat:${getNumericWeights(BODY_WEIGHTS)}`,
].join('');
