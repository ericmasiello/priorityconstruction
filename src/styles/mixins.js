import { pxToRem } from './utils';
import { BODY_FONT_STACK } from './vars';
import blueprint from '../images/blueprint.svg';
import grid from '../images/grid.svg';

export const baseFont = () => `font-family: ${BODY_FONT_STACK}`;
export const headerFont = () => `font-family: ${BODY_FONT_STACK}`;

export const type = typeDef =>
  `
    font-size: ${pxToRem(typeDef[0])};
    line-height: ${typeDef[1]};
  `;

export const scalableType = ([maxSize, lineHeight], minSize = 16) => {
  const relativeScaler = 10;
  return `
    font-size: 10vw;
    line-height: ${lineHeight};

    @media(max-width: ${pxToRem(minSize * relativeScaler)}) {
      font-size: ${pxToRem(minSize)};
    }

    @media(min-width: ${pxToRem(maxSize * relativeScaler)}) {
      font-size: ${pxToRem(maxSize)};
    }
  `;
};

export const siteBg = () => `
  min-height: 100vh;
  background-image: 
    url(${blueprint}),
    linear-gradient(to bottom, rgba(255, 255, 255, 0.6) 0%,rgba(255, 255, 255, 0.6) 100%),
    url(${grid});
  background-size:
    contain,
    100%,
    ${pxToRem(960)};
  background-repeat: 
    no-repeat,
    repeat,
    repeat;
  background-position: bottom left;
  display: grid;
  grid-template-rows: 1fr auto;
`;
