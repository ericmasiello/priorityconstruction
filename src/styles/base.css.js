import { pxToRem } from './utils';
import { baseFont, headerFont } from './mixins';
import { COLORS, BODY_WEIGHTS, FONT_URL, scaler } from './vars';

export default `
@import url('${FONT_URL}');

  html {
    box-sizing: border-box;
    background-color: ${COLORS.bg};
  }

  *, *::before, *::after {
    box-sizing: inherit;
  }

  html {
    ${baseFont()};
    color: ${COLORS.base};
    font-size: 100%;
  }

  body {
    margin: 0;
    font-size: ${pxToRem(16)};
    line-height: ${scaler};
    font-weight: ${BODY_WEIGHTS.regular};
  }

  input, textarea, select, button {
    font-size: inherit;
    font-family: inherit;
    font-weight: inherit;
    color: inherit;
  }

  label,
  input[type=button],
  input[type=submit],
  input[type=reset],
  button {
    cursor: pointer;
  }

  h1, h2, h3, h4, h5, h6 {
    ${headerFont()};
    font-weight: ${BODY_WEIGHTS.regular};
    line-height: 1.1;
  }

  h1, h2, h3, h4, h5, h6, p {
    margin-top: 0;
    margin-bottom: 1rem;
  }

  a {
    background-color: transparent;
    color: ${COLORS.link};
    text-decoration: none;
    -webkit-tap-highlight-color: transparent;
  }

  abbr {
    position: relative;
    text-decoration: none;

    &::after {
      content: '';
      position: absolute;
      z-index: -1;
      width: 100%;
      height: ${pxToRem(2)};
      left: 0;
      bottom: ${pxToRem(-3)};
      background-color: ${COLORS.base};
      opacity: 0.3;
    }
  }

  img, svg {
    max-width: 100%;
    height: auto;
  }

  img {
    vertical-align: middle;
    border: 0;
  }

  blockquote {
    margin: 0;
  }

  hr {
    border-width: 1px 0 0 0;
    border-style: solid;
  }

  address {
    font-style: normal;
  }

  fieldset {
    border: none;
    padding: 0;
    margin: 0;
  }
`;
