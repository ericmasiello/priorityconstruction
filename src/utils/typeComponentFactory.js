import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { type, scalableType } from '../styles/mixins';
import { BODY_WEIGHTS } from '../styles/vars';

const typeComponentFactory = (
  typeSize,
  { defaultTag = 'div', displayName = `Type(${typeSize})` } = {},
) => {
  const Type = props => {
    const { tag: Tag, scale, uppercase, bold, ...rest } = props;
    return <Tag {...rest} />;
  };

  Type.defaultProps = {
    tag: defaultTag,
    scale: true,
  };

  Type.propTypes = {
    tag: PropTypes.oneOfType([PropTypes.func, PropTypes.string]),
    className: PropTypes.string,
    scale: PropTypes.bool,
    uppercase: PropTypes.bool,
    bold: PropTypes.bool,
  };

  Type.displayName = displayName;

  // TODO: add code for making type capitlize and bold
  const StyledType = styled(Type)`
    ${props => {
      const { scale, bold, uppercase } = { ...Type.defaultProps, ...props };
      const styles = [scale ? scalableType(typeSize) : type(typeSize)];

      if (bold) {
        styles.push(`font-weight: ${BODY_WEIGHTS.bold};`);
      }

      if (uppercase) {
        styles.push(`text-transform: uppercase;`);
      }

      return styles.join('');
    }};
  `;

  return {
    Type,
    StyledType,
  };
};

export default typeComponentFactory;
