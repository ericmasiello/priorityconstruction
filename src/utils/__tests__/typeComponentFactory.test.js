import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import { shallow } from 'enzyme';
import typeComponentFactory from '../typeComponentFactory';
import { TYPE_SIZE } from '../../styles/vars';

it('should create a renderable styled component', () => {
  const Wrapper = typeComponentFactory(TYPE_SIZE.t1);
  const component = renderer.create(<Wrapper.StyledType />);

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it('should create a renderable component', () => {
  const Wrapper = typeComponentFactory(TYPE_SIZE.t1);
  const component = renderer.create(<Wrapper.Type />);

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it('should set the text to uppercase', () => {
  const Wrapper = typeComponentFactory(TYPE_SIZE.t1);
  const component = renderer.create(<Wrapper.StyledType uppercase />);

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it('should set the text to bold', () => {
  const Wrapper = typeComponentFactory(TYPE_SIZE.t1);
  const component = renderer.create(<Wrapper.StyledType weight="bold" />);

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it('should render a <div /> by default', () => {
  const Wrapper = typeComponentFactory(TYPE_SIZE.t1);
  const wrapper = shallow(<Wrapper.Type />);

  expect(wrapper.type()).toBe('div');
});

it('should allow the default tag to be customized', () => {
  const Wrapper = typeComponentFactory(TYPE_SIZE.t1, { defaultTag: 'span' });
  const wrapper = shallow(<Wrapper.Type />);

  expect(wrapper.type()).toBe('span');
});

it('should set custom displayName', () => {
  const Wrapper = typeComponentFactory(TYPE_SIZE.t1, { displayName: 'Foo' });

  expect(Wrapper.Type.displayName).toBe('Foo');
});
