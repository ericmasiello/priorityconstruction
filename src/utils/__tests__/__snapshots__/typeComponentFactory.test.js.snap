// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should create a renderable component 1`] = `<div />`;

exports[`should create a renderable styled component 1`] = `
.c0 {
  font-size: 10vw;
  line-height: 1.1;
}

@media (max-width:10rem) {
  .c0 {
    font-size: 1rem;
  }
}

@media (min-width:31.57334518321rem) {
  .c0 {
    font-size: 3.157334518321rem;
  }
}

<div
  className="c0"
/>
`;

exports[`should set the text to bold 1`] = `
.c0 {
  font-size: 10vw;
  line-height: 1.1;
  font-weight: 600;
}

@media (max-width:10rem) {
  .c0 {
    font-size: 1rem;
  }
}

@media (min-width:31.57334518321rem) {
  .c0 {
    font-size: 3.157334518321rem;
  }
}

<div
  className="c0"
/>
`;

exports[`should set the text to uppercase 1`] = `
.c0 {
  font-size: 10vw;
  line-height: 1.1;
  text-transform: uppercase;
}

@media (max-width:10rem) {
  .c0 {
    font-size: 1rem;
  }
}

@media (min-width:31.57334518321rem) {
  .c0 {
    font-size: 3.157334518321rem;
  }
}

<div
  className="c0"
/>
`;
